packages/@aws-cdk/cloudformation-include/test/nested-stacks.test.ts

```diff
@@ -611,6 +611,43 @@ describe('CDK Include', () => {
         loadTestFileToJsObject('grandchild-import-stack.json'),
       );
     });
+
+    test('parameters are replaced across nested stacks', () => {
+      assetStack = new core.Stack();
+      parentTemplate = new inc.CfnInclude(assetStack, 'ParentStack', {
+        templateFile: testTemplateFilePath('parent-one-child.json'),
+        nestedStacks: {
+          'ChildStack': {
+            templateFile: testTemplateFilePath('child-import-stack.json'),
+            parameters: {
+              'MyBucketParameter': 'bucket-name',
+            },
+          },
+        },
+      });
+
+      child = parentTemplate.getNestedStack('ChildStack');
+      expect(child.stack).toMatchTemplate({
+        "Resources": {
+          "BucketImport": {
+            "Type": "AWS::S3::Bucket",
+            "Properties": {
+              "BucketName": "bucket-name",
+            },
+          },
+          "GrandChildStack": {
+            "Type": "AWS::CloudFormation::Stack",
+            "Properties": {
+              "TemplateURL": "https://cfn-templates-set.s3.amazonaws.com/grandchild-import-stack.json",
+              "Parameters": {
+                // not a reference, so it isn't replaced
+                "MyBucketParameter": "some-other-bucket-name",
+              },
+            },
+          },
+        },
+      });
+    });
```

Comment:
Missing assertion from the parent stack that `MyBucketParameter` is no longer passed to `ChildStack`.